pkg/utils/openshift/webconsole_test.go

```diff
@@ -0,0 +1,84 @@
+package openshift
+
+import (
+	"github.com/ghodss/yaml"
+	oappsv1 "github.com/openshift/api/apps/v1"
+	v1 "github.com/openshift/api/console/v1"
+	"github.com/stretchr/testify/assert"
+	appsv1 "k8s.io/api/apps/v1"
+	"testing"
+)
+
+func TestGetConsoleYAMLSample(t *testing.T) {
+	var inputYaml = `
+apiVersion: v1
+kind: DeploymentConfig
+metadata:
+ name: sample-dc
+ annotations:
+   consoleName: sample-deploymentconfig
+   consoleDesc: Sample Deployment Config
+   consoleTitle: Sample Deployment Config
+spec:
+   replicas: 2
+`
+	resource := &oappsv1.DeploymentConfig{}
+	assert.NoError(t, yaml.Unmarshal([]byte(inputYaml), resource))
+
+	yamlSample, err := GetConsoleYAMLSample(resource)
+	assert.NoError(t, err)
+
+	assert.Equal(t, "sample-deploymentconfig", yamlSample.ObjectMeta.Name)
+	assert.Equal(t, "openshift-console", yamlSample.ObjectMeta.Namespace)
+	assert.Equal(t, "v1", yamlSample.Spec.TargetResource.APIVersion)
+	assert.Equal(t, "DeploymentConfig", yamlSample.Spec.TargetResource.Kind)
+	assert.Equal(t, v1.ConsoleYAMLSampleTitle("Sample Deployment Config"), yamlSample.Spec.Title)
+	assert.Equal(t, v1.ConsoleYAMLSampleDescription("Sample Deployment Config"), yamlSample.Spec.Description)
+
+	yamlContent := yamlSample.Spec.YAML
+	res := &appsv1.Deployment{}
+	assert.NoError(t, yaml.Unmarshal([]byte(string(yamlContent)), res))
+	assert.Empty(t, res.Annotations, "Annotations should be cleaned up.")
+	assert.Equal(t, "sample-dc", res.ObjectMeta.Name)
```

Comment:
Instead of the next few lines, can we not compare res with resource, and shouldn't the two objects be equal?